### ISSUE TITLE
terraform.enable option

### ISSUE THREAD

**Issue by arcnmx** [arcnmx/tf-nix#8]
Expose a way to tell whether a build/evaluation is meant to include tf-nix resources or not. This basically means it's in one of the following scenarios:
1. it's a bare nix build, but tfstate exists (thus `tf.state.enable == true`)
    - tfstate could be incomplete though, should this distinction be considered?
2. tfstate does not exist (yet), but it's a recursive `terraform apply` command build.
    - note that if this is an `import` operation across terraform targets, we may be "in" tf-nix execution but can't expect the state will be updated by the next iteration (or ever)
    - counternote: maybe such a dependency *should* actually trigger an apply on the dependent target, then you can!

So basically, if `tf.enable` is true, the config can rely on the fact that the resource it needs will be available eventually, even if not in the current evaluation. It should still use the resource string so that `deps` can note that it's a dependency. If false (say we're in a CI build with no state), there is no hope the state will ever exist and it should omit it from the evaluation entirely.
